projects/RabbitMQ.Client/Impl/RabbitMQActivitySource.cs

```diff
@@ -56,6 +67,20 @@ public static class RabbitMQActivitySource
             new KeyValuePair<string, object?>(ProtocolVersion, "0.9.1")
         };
 
+        internal static Activity? OpenConnection(IFrameHandler frameHandler)
+        {
+            if (!s_connectionSource.HasListeners())
+            {
+                return null;
+            }
+
```

Comment:
```suggestion
```

I don't think this is necessary. `ActivitySource.CreateActivity` will do this check and return null.

https://github.com/dotnet/runtime/blob/1622f514684d94a521bfb41c88a27079ad943ee7/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/ActivitySource.cs#L192-L200

Reply:
I take the same logic as the rest of the implem but you are right when I read the Microsoft code this line is useless